File: app/src/main/java/com/varunbarad/popularmovies/eventlistener/OnFragmentInteractionListener.java
package com.varunbarad.popularmovies.eventlistener;

/**
 * Creator: vbarad
 * Date: 2016-12-04
 * Project: PopularMovies
 */

public interface OnFragmentInteractionListener {
  String TAG_MOVIE = "MOVIE";
  String TAG_FAVORITE = "FAVORITE";
  String TAG_UNFAVORITE = "UNFAVORITE";
  
  void onFragmentInteraction(String message);
}

File: app/src/main/java/com/varunbarad/popularmovies/eventlistener/OnFragmentInteractionListener.kt
package com.varunbarad.popularmovies.eventlistener

import com.varunbarad.popularmovies.model.MovieStub

/**
 * Creator: Varun Barad
 * Date: 2019-06-04
 * Project: PopularMovies
 */
sealed class FragmentInteractionEvent {
    class OpenMovieDetailsEvent(val movie: MovieStub): FragmentInteractionEvent()
    class AddToFavoriteEvent(val movie: MovieStub): FragmentInteractionEvent()
    class RemoveFromFavoriteEvent(val movieId: Long): FragmentInteractionEvent()
}
